.github/workflows/ci.yaml

```diff
@@ -250,6 +250,13 @@ jobs:
         with:
           artifact-name: kind-logs-${{ matrix.ip-family }}-${{ matrix.bgp-type }}-${{ matrix.deployment}}
 
+      - name: Publish JUnit Report
+        uses: mikepenz/action-junit-report@v4
+        if: success() || failure()
```

Comment:
do we need this if?

Reply:
This job will run whether the E2E test succeeds or fails.

Reply:
yep but, since we publish in both cases, maybe the if is not needed at all?

Reply:
I'm not sure. I'd be worried that the job would not be executed because the previous job failed.

Reply:
@fedepaol I'm sure it's needed.
Test result: If this line is not added, it will be skipped.
<img width="288" alt="image" src="https://github.com/metallb/metallb/assets/17496418/ba24171d-a3f8-46c6-9eda-6543af2f0ae4">

This is my test code:
```
jobs:
  # This workflow contains a single job called "build"
  build:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
      - name: Run a one-line script
        run: echo Hello, world!
      - name: Try to fail
        run: exit 1
      - name: Print message if we don't fail
        run: echo Should not get here
      - name: Run a multi-line script
        if: success() || failure()
        run: |
          echo Add other actions to build,
          echo test, and deploy your project.
```

Reply:
thanks for checking!